src/parser/json/annotation_parser.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2022 spdx contributors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from datetime import datetime
+from typing import Dict, Optional, List
+
+from src.model.actor import Actor
+from src.model.annotation import Annotation, AnnotationType
+from src.parser.error import SPDXParsingError
+from src.parser.json.actor_parser import ActorParser
+from src.parser.json.dict_parsing_functions import datetime_from_str, construct_or_raise_parsing_error, \
+    parse_field_or_log_error, append_parsed_field_or_log_error, raise_parsing_error_if_logger_has_messages
+from src.parser.logger import Logger
+
+
+class AnnotationParser:
+    logger: Logger
+    actor_parser: ActorParser
+
+    def __init__(self):
+        self.logger = Logger()
+        self.actor_parser = ActorParser()
+
+    def parse_all_annotations(self, input_doc_dict: Dict) -> List[Annotation]:
+        annotations_list = []
+        self.parse_annotations_from_object(annotations_list, [input_doc_dict])
+        reviews: List[Dict] = input_doc_dict.get("revieweds")
```

Comment:
You are right, and thanks for the suggestion with the empty list, I totally forgot that one can also specify a return value for `get` if the key doesn't exist. This really improves the code regarding all the `if ..`. 
Regarding the optional type hints, I wasn't sure as I would get the impression that the corresponding value is optional, so e.g. `input_doc_dict.get("SPDXID")` isn't optional in the data model, but of course this could return `None`. I guess, I was somehow looking from the wrong perspective as the type hint should match the possible return not the intended value (especially as this is caught by the constructor in the end...)